exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/TextFormatPlugin.java

```diff
@@ -81,9 +68,8 @@
  * to allow tight control of the size of produced batches (as well
  * as to support provided schema.)
  */
-public class TextFormatPlugin extends EasyFormatPlugin<TextFormatPlugin.TextFormatConfig> {
-
-  private final static String PLUGIN_NAME = "text";
+public class TextFormatPlugin extends EasyFormatPlugin<TextFormatConfig> {
+  final static String PLUGIN_NAME = "text";
```

Comment:
I know this would probably break things, but I'll ask anyway.  Do you think this plugin would be better named `delimited`?  It really refers to delimited formats such as PSV, TSV, etc.   If this is a major breaking change, don't worry about it.  It's always bothered me though.

Reply:
Drill was originally designed for big data: huge CSV files that get split across readers. That huge file might have a header, but more typically did not. The CSV reader still handles both cases: if the file has a header, it will seek to the beginning of the file even if the "file work" says to start at an offset of 256MB, etc.

In your use case, you likely work with "small data" but, with much variety and you want to use Drill as a convenient way to work with such data sets. (Hence all the recent work on the REST API which is decidedly *not* a big data solution!)

If we change the default, then any "big data" users will find their apps break if they still happen to use CSV files without headers.

This might be something the team chooses to change in Drill 2.0 along with other breaking changes: shift the focus from old-style HDFS files to modern S3 files are your case of small data-science style files.

Reply:
I wonder if it might make sense to follow the model of some of the other format readers where if the column headers are not known, it assigns a name of `field_n` to them.  That way you still get columns instead of the columns array.  

My view is that if we change the default, but still allow the users to revert back with a config change, it's not that big of a deal.

Reply:
@cgivre @paul-rogers the master branch is now rolling towards 2.0.  If we don't take the chance in this cycle to improve user-facing stuff, it's a long time until the next chance.

I do think we should try to segregate our breaking changes into their own PRs, and therefore squashed commits, as much as we can so that our ability to cherry pick back to a 1.20.1 or, god forbid, a 1.21 is maxmimally preserved.

FWIW, I'm +1 for turning `columns[n]` into `column_n` (and I think this terminolgy is more standard for SQL result sets than `field`), +1 for "text" becoming "delimited" and +0 for the default value of `extractHeaders`, which is trivial for users to set for themselves.

Reply:
The team can certainly many things in the 2.0 branch. But, this is not a 2.0 PR. The goal of this PR is to preserve functionality and simply upgrade the underlying reader mechanism to use EVF 2.

Reply:
On the `field_x` vs `columns[]` issue, that part is possible in the current code, as an option. There is a "columns" class in EVF that handles the `columns[]` column. I suspect that a parallel "fields" version could be created that generates the numbered fields instead of array indexes. An option on the plugin would choose one or the other for the no-headers case.

In fact, with this approach, the individual `field_n` columns could be typed, based on first-row type inference: something not possible with `columns[]`.

Anyone up for giving it a shot? I'd be happy to offer pointers along the way.